### PR TITLE
[Flight] Move around the Server side a bit

### PR DESCRIPTION
**Pull request by sebmarkbage**
__Builds on top of #17234.__

Just some clean up before I add more stuff.

### CODE REVIEW

File: packages/react-dom/src/__tests__/ReactFlightDOM-test.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      return model;
+    }
+
+    let {writable, readable} = getTestStream();
+    ReactFlightDOMServer.pipeToNodeWritable(<App />, writable);
+    let result = ReactFlightDOMClient.readFromReadableStream(readable);
+    await waitForSuspense(() => {
+      expect(result.model).toEqual({
+        html: '<div><span>hello</span><span>world</span></div>',
+      });
```
Comment (gaearon) [inline packages/react-dom/src/__tests__/ReactFlightDOM-test.js]: Mind asserting we actually eventually got here?

File: packages/react-dom/src/__tests__/ReactFlightDOMBrowser-test.js [hunk 2]
```diff
@@ -48,14 +56,19 @@ describe('ReactFlightDOM', () => {
...
+      };
+      return model;
+    }
+
+    let stream = ReactFlightDOMServer.renderToReadableStream(<App />);
+    let result = ReactFlightDOMClient.readFromReadableStream(stream);
+    await waitForSuspense(() => {
+      expect(result.model).toEqual({
+        html: '<div><span>hello</span><span>world</span></div>',
+      });
```
Comment (gaearon) [inline packages/react-dom/src/__tests__/ReactFlightDOMBrowser-test.js]: Same

File: packages/react-dom/src/__tests__/ReactFlightDOM-test.js (hunk 1 above)
Comment (sebmarkbage) [reply packages/react-dom/src/__tests__/ReactFlightDOM-test.js]: There’s going to be a hundred of these. I’m not going to assert in each one that the testing environment works.

Jest fails it the promise to the test never resolves.

Comment (gaearon) [reply packages/react-dom/src/__tests__/ReactFlightDOM-test.js]: I mean the promise could resolve without the inner expect actually having run successfully. If there’s a bug in waitForSuspense.

**Pull request merged**